protocol/contracts/libraries/Silo/LibLegacyTokenSilo.sol

```diff
@@ -0,0 +1,600 @@
+                    perDepositData.season
+                );
+ 
+                // also need to calculate how much stalk has grown since the migration
+                uint128 stalkGrownSinceStemStartSeason = LibSilo.stalkReward(0, perTokenData.stemTip, crateBDV.toUint128()).toUint128();
+                perDepositData.grownStalk = perDepositData.grownStalk.add(stalkGrownSinceStemStartSeason);
+                migrateData.totalGrownStalk = migrateData.totalGrownStalk.add(stalkGrownSinceStemStartSeason);
+ 
+                // add to new silo
+                LibTokenSilo.addDepositToAccount(
+                    account, 
+                    perTokenData.token, 
+                    LibTokenSilo.grownStalkAndBdvToStem(
+                        IERC20(perTokenData.token), 
+                        perDepositData.grownStalk,
+                        crateBDV
+                    ), 
+                    perDepositData.amount, 
+                    crateBDV,
+                    LibTokenSilo.Transfer.emitTransferSingle
+                );
+ 
+                // add to running total of seeds
+                migrateData.totalSeeds = migrateData.totalSeeds.add(crateBDV.mul(getSeedsPerToken(address(perTokenData.token))).toUint128());
+
+                // emit legacy RemoveDeposit event
+                emit RemoveDeposit(account, perTokenData.token, perDepositData.season, perDepositData.amount);
+            }
+ 
+            // init mow status for this token
+            setMowStatus(account, perTokenData.token, perTokenData.stemTip);
+        }
+ 
+        // user deserves stalk grown between stemStartSeason and now
+        LibSilo.mintGrownStalk(account, migrateData.totalGrownStalk);
+
+        //return seeds diff for checking in the "part 2" of this function (stack depth kept it from all fitting in one)
+        return balanceOfSeeds(account).sub(migrateData.totalSeeds);
+    }
+
+    function _mowAndMigrateMerkleCheck(
+        address account,
+        uint256 stalkDiff,
+        uint256 seedsDiff,
+        bytes32[] calldata proof,
+        uint256 seedsVariance
+    ) internal {
+        if (seedsDiff > 0) {
+            //verify merkle tree to determine stalk/seeds diff drift from convert issue
+            bytes32 leaf = keccak256(abi.encode(account, stalkDiff, seedsDiff));
+            
+            require(
+                MerkleProof.verify(proof, DISCREPANCY_MERKLE_ROOT, leaf),
+                "UnripeClaim: invalid proof"
+            );
+        }
+
+        //make sure seedsVariance equals seedsDiff input
+        require(seedsVariance == seedsDiff, "seeds misalignment, double check submitted deposits");
+
+        AppStorage storage s = LibAppStorage.diamondStorage();
+
+
+        //and wipe out old seed balances (all your seeds are belong to stem)
+        setBalanceOfSeeds(account, 0);
+
+        //emit that all their seeds are gone
+        emit SeedsBalanceChanged(account, -int256(s.a[account].s.seeds));
```

Comment:
Should this happen before line 436? If it sets the Seeds to zero before emitting the event, then `-int256(s.a[account].s.seeds` will be zero.

Reply:
Yes, fixed here, will work on a seeds test as well: https://github.com/BeanstalkFarms/Beanstalk/commit/4fb20bc7bad96a29570a6e14c352a86c1ba0e478